src/validator/P256Validator.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import {UserOperation} from "I4337/interfaces/UserOperation.sol";
+import {ECDSA} from "solady/utils/ECDSA.sol";
+import {IKernelValidator} from "../interfaces/IKernelValidator.sol";
+import {ValidationData} from "../common/Types.sol";
+import {SIG_VALIDATION_FAILED} from "../common/Constants.sol";
+import {P256} from "p256-verifier/P256.sol";
+
+contract P256Validator is IKernelValidator {
+    event P256PublicKeysChanged(address indexed kernel, PublicKey indexed oldKeys, PublicKey indexed newKeys);
```

Comment:
Why indexing the `oldKeys` & `newKeys`? It will increase gas usage. 
And oldKeys isn't rly needed here I think, it will prevent two sload (for x & y), during module enabling